packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -732,9 +735,19 @@ export async function createRouteManifest(
 		}
 	}
 
-	if (!dev) {
-		await runHookRoutesResolved({ routes, settings, logger });
+	if (dev) {
+		// In SSR, a 404 route is injected in the App directly for some special handling,
+		// it must not appear in the manifest
+		ensure404Route({ routes });
 	}
+	if (dev || settings.buildOutput === 'server') {
+		injectImageEndpoint(settings, { routes }, dev ? 'dev' : 'build');
+		// During the build build, we can't only inject this route when server islands are used because:
```

Comment:
this comments a little confusing.

I'm not sure what happens today if you have a static project with only server islands. Does it generate a server build or not? If it does, then I think this change will break it. If not, that's a bug but probably not one that this affects.

Could you test this and confirm what happens?

Reply:
My understanding is that you cannot build a static project that has server islands, see https://github.com/withastro/astro/blob/feat/route-manifest-refactor/packages/astro/src/core/build/index.ts#L213-L215. So having server islands does bot make the build output server, so it should be safe in this case

Reply:
Yeah, I can confirm that. I tested it right now. Having a server island without an adapter will result in an error.